common/src/stack/pylib/stack/util.py

```diff
@@ -171,3 +172,12 @@ def unique_everseen(iterable, key=None):
 def lowered(iterable):
 	"""Return a generator that lowercases all strings in the provided iterable."""
 	return (string.lower() for string in iterable)
+
+def is_valid_hostname(name):
+	"""Check if a given name is a valid hostname.
+
+	For our purposes, a valid hostname is a single hostname label (or "name")
+	as defined by RFCs 952 and 1123.
+	"""
+	pattern = r"^[a-zA-Z0-9]$|^[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]$"
```

Comment:
Since we always never care about case, leave the different case classes out of the pattern and just pass `match` the case insensitive flag. Should make the regex easier to read. Also I'd prefer a comment explaining (briefly) what the regex is intended to match, in case someone decides to muck with it later. The `re.verbose` flag might be useful there.

Reply:
Incorporated in [e48daf4](https://github.com/Teradata/stacki/pull/641/commits/e48daf47da978c27cce42be7b97408e498c5b4a1).